daemon/cmd/hubble.go

```diff
@@ -137,10 +137,36 @@ func (d *Daemon) launchHubble() {
 		logger.WithFields(logrus.Fields{
 			"address": option.Config.HubbleMetricsServer,
 			"metrics": option.Config.HubbleMetrics,
+			"tls":     option.Config.HubbleMetricsServerTLSEnabled,
 		}).Info("Starting Hubble Metrics server")
 		grpcMetrics := grpc_prometheus.NewServerMetrics()
 
-		if err := metrics.EnableMetrics(log, option.Config.HubbleMetricsServer, option.Config.HubbleMetrics, grpcMetrics, option.Config.EnableHubbleOpenMetrics); err != nil {
+		var metricsTLSConfig *certloader.WatchedServerConfig
+		if option.Config.HubbleMetricsServerTLSEnabled {
+			metricsTLSConfigChan, err := certloader.FutureWatchedServerConfig(
+				logger.WithField("config", "hubble-metrics-server-tls"),
+				option.Config.HubbleMetricsServerTLSClientCAFiles,
+				option.Config.HubbleMetricsServerTLSCertFile,
+				option.Config.HubbleMetricsServerTLSKeyFile,
+			)
+			if err != nil {
+				logger.WithError(err).Error("Failed to initialize Hubble metrics server TLS configuration")
+				return
+			}
+			waitingMsgTimeout := time.After(30 * time.Second)
+			for metricsTLSConfig == nil {
+				select {
+				case metricsTLSConfig = <-metricsTLSConfigChan:
+				case <-waitingMsgTimeout:
+					logger.Info("Waiting for Hubble metrics server TLS certificate and key files to be created")
+				case <-d.ctx.Done():
+					return
```

Comment:
Should we add a log msg here about the cause of hubble launch failure?

Reply:
💯  we can do that in the existing hubble server TLS setup too.